test/new-e2e/pkg/e2e/suite.go

```diff
@@ -510,11 +510,13 @@ func (bs *BaseSuite[Env]) SetupSuite() {
 		// run environment diagnose
 		if diagnosableEnv, ok := any(bs.env).(common.Diagnosable); ok {
```

Comment:
You can just do this to avoid adding one indentation level below
```suggestion
		if diagnosableEnv, ok := any(bs.env).(common.Diagnosable); ok && diagnosableEnv != nil {
```